File: cPlusPlusCode/gammaAdjustment.cpp
#include<opencv2/imgproc.hpp>
#include<opencv2/highgui.hpp>
#include<iostream>

using namespace std;
using namespace cv;

int main(){

    //Read image
    Mat image = imread("../assets/putin.jpg");

    //define gamma
    float gamma = 1.5;

    Mat lookupTable(1, 256, CV_8U);
    uchar* lut = lookupTable.ptr();
    for(int i=0; i<256; i++){
        lut[i] = int(255*(pow((float)i/255.0, gamma)));
    }
    cout<<lookupTable<<endl;

    //gamma adjustment. convert image using LUT table. It maps the pixel intensities in the input to the output using values from lut
    Mat result;
    LUT(image, lookupTable, result);

    //create windows to display images
    namedWindow("image", WINDOW_NORMAL);
    namedWindow("result", WINDOW_NORMAL);

    //display images
    imshow("image", image);
    imshow("result", result);

    //press esc to exit the prgoram
    waitKey(0);

    //close all the opened windows
    destroyAllWindows();

    return 0;
}

File: pythonCode/gammaAdjustment.py
import cv2
import numpy as np

#read image
image = cv2.imread("../assets/putin.jpg")

#define gamma
gamma = 1.5

#create lookup table
values = np.arange(0, 256)
lut = np.uint8(255 * np.power((values/255.0), gamma))
print(lut)

#gamma adjustment. convert image using LUT table. It maps the pixel intensities in the input to the output using values from lut
result = cv2.LUT(image, lut)

#create windows to display image
cv2.namedWindow("image", cv2.WINDOW_NORMAL)
cv2.namedWindow("result", cv2.WINDOW_NORMAL)

#display images
cv2.imshow("image", image)
cv2.imshow("result", result)

#press Esc to exit the program
cv2.waitKey(0)

#close all the opended windows
cv2.destroyAllWindows()